### ISSUE TITLE
3D image location

### ISSUE THREAD

**Issue by markgrujic** [YaleDHLab/pix-plot#71]
Hi,
Thanks for this fantastic code.

I was wondering, is it possible to change the number of dimensions in the UMAP reduction to 3, and then visualise the 3D distribution instead of the 2D plane with parallax?

Cheers

**Comment by duhaime** [comment]
@markgrujic many thanks for your note. It's certainly possible to achieve that effect. In fact, @hadro did just that here: https://github.com/hadro/pix-plot-dpla

Would you want to give that code a quick study (or try to run it out of the box) to achieve this effect?

If you get stuck or need some step by step guidance we can certainly follow up!

**Comment by hadro** [comment]
Hi, yes, I added `n_components=3` to the umap parameters in utils/process_images.py:
https://github.com/hadro/pix-plot-dpla/blob/master/utils/process_images.py#L239

I also had to update the `get_image_positions(self, fit_model)` function to return z coordinates in addition to x and y.

After that, I had to update the javascript to plot the z dimension instead of the random bounded value that creates slight jitter in the 2d plane. Hope that all makes sense.
It's been a while since I did all that, but hope it helps!
